lib/ex_machina.ex

```diff
@@ -96,6 +100,32 @@ defmodule ExMachina do
   end
 
   @doc """
+  Builds a factory with the passed in factory_name and returns its fields
+
+  ## Example
+
+      def factory(:user) do
+        %MyApp.User{name: "John Doe", admin: false}
+      end
+
+      # Returns %{name: "John Doe", admin: true}
+      fields_for(:user, admin: true)
+  """
+  def fields_for(module, factory_name, attrs \\ %{}) do
+    module.build(factory_name, attrs)
+    |> drop_undesireable_fields
+  end
+
+  defp drop_undesireable_fields(record = %{__struct__: struct}) do
```

Comment:
This is pretty Ecto specific. Should this be called `drop_unwanted_ecto_fields`?